File: ITask.py
# coding: UTF-8

# interface接口: 任务接口
class ITask():
    def __init__(self):

    """
    注入用户参数内容
    Returns:
        ArgumentParser 回传传入的解析器对象
    """
    def UserArgumentAdd(self, keyname, parser):
        return parser

    """
    执行相关的任务操作
    Returns:
        repo 回传传入的仓库相关对象
    """
    def OnExecute(self, repo):
        return repo

    """
    输出打印结果
    Returns:
        string[] 打印结果
    """
    def PrintResult(self, repo):
        return []

File: src/RunCommand.Logic/ITask.cs
using System.Collections.Generic;

namespace RunCommand
{
    /// <summary>
    /// 执行任务
    /// </summary>
    public interface ITask
    {
        /// <summary>
        /// 执行任务内容
        /// </summary>
        /// <param name="repo">存储库</param>
        /// <returns>回传存储库信息</returns>
        IRepository OnExecute(IRepository repo);

        /// <summary>
        /// 输出打印结果
        /// </summary>
        /// <param name="repo">存储库</param>
        /// <returns>打印结果</returns>
        IList<string> PrintResult(IRepository repo);

        /// <summary>
        /// 是否将输出分割
        /// </summary>
        /// <param name="repo">存储库</param>
        /// <returns>结果</returns>
        bool IsWriteIntervalLine(IRepository repo);
    }
}
